### PR TITLE
Summarizer API and Gallery API

### PR DESCRIPTION
**Pull request by NAshwinKumar**

### CODE REVIEW

File: src/app/deep-read/.gitignore [hunk 1]
```diff
@@ -144,6 +144,6 @@ model*/
 *.avi
 *.mp3
 
-# transcript
-core/temp/.*
-core/temp/*json
+# Temporary files for transcript and summary
+core/temp/
+!core/temp/README.md
```
Comment (Acquil) [inline src/app/deep-read/.gitignore:148]: Empty files don't get added to git. If we clone and run, there will be an error saying no such directory.

File: src/app/deep-read/azure_blob_settings.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+from os import environ
+CONNECT_STR = environ.get('BLOB_CONNECTION_STR')
```
Comment (Acquil) [inline src/app/deep-read/azure_blob_settings.py]: settings.py can have these variables

File: src/app/deep-read/.gitignore (hunk 1 above)
Comment (NAshwinKumar) [reply src/app/deep-read/.gitignore:148]: I have added readme.md as an exception in last line, that will make sure that folder is created right?

File: src/app/deep-read/core/video_to_text/video_to_text_converter.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        self.render_path = os.path.splitext(self.video_path)[0] #rendered
+        self.offset_duration_seconds = offset_duration_seconds
+    
+    def extract_images_from_video(self):
+        video_to_jpg_converter = Video_to_JPG_Converter(self.video_path, self.render_path, self.offset_duration_seconds)
+        video_to_jpg_converter.convert()
+    
+    #Images are classified as screens or slides(presentation slides) using NN, we will convert only slide images into text
+    def get_slide_images_from_extracted_images(self):
+        self.slide_images = Video_to_Text_Converter.classifier.predict("/media/ashwin/Current/sem_9/open_source/deep-read/src/app/deep-read/core/temp/sample_transcript1")
```
Comment (Acquil) [inline src/app/deep-read/core/video_to_text/video_to_text_converter.py]: path is absolute here

File: src/app/deep-read/cuda_installation_script.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+# Add NVIDIA package repositories
+wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-repo-ubuntu1804_10.1.243-1_amd64.deb
```
Comment (Acquil) [inline src/app/deep-read/cuda_installation_script.sh]: Is there a way to get this instead of specifying 'ubuntu18'?

**Review by Acquil — CHANGES_REQUESTED** [review]
Needs updates.

File: src/app/deep-read/.gitignore (hunk 1 above)
Comment (Acquil) [reply src/app/deep-read/.gitignore:148]: 👍

File: src/app/deep-read/azure_blob_settings.py (hunk 2 above)
Comment (Acquil) [reply src/app/deep-read/azure_blob_settings.py]: Done

File: src/app/deep-read/core/video_to_text/video_to_text_converter.py (hunk 3 above)
Comment (NAshwinKumar) [reply src/app/deep-read/core/video_to_text/video_to_text_converter.py]: fixed this bug in [e8f6d24](https://github.com/Acquil/deep-read/pull/32/commits/e8f6d24f2f6d31c30db0df37abe1e4916adab56e)

**Review by Acquil — APPROVED** [review]
Done

**Pull request merged**